lib/tern.js

```diff
@@ -749,8 +749,14 @@
       if (expr) return expr;
       expr = infer.findExpressionAround(file.ast, start, end, file.scope);
       if (expr && (expr.node.type == "ObjectExpression" || wide ||
-                   (start == null ? end : start) - expr.node.start < 20 || expr.node.end - end < 20))
+                   (start == null ? end : start) - expr.node.start < 20 || expr.node.end - end < 20)) {
+        if (expr.node.type == "ObjectExpression" ) {
+          // return object literal property if possible.
+          var objProp = pointInProp(expr.node, end);
+          if (objProp) return {node: objProp, state: expr.state};          
```

Comment:
I have created this PR to try to implement find type for property key, but it seems that it's not the best mean since it breaks tests. I hope you will accept this feature by implementing it with a clean mean.